lib/pages/qr_scanner.dart

```diff
@@ -0,0 +1,112 @@
+              //This pushes back an argument through the route on back button press
+              onPressed: () => Navigator.pop(context, passToMainScreen)),
+          title: Text(
+            widget.title ?? '',
+            style: TextStyle(color: Theme.of(context).accentColor),
+          ),
+          backgroundColor: Theme.of(context).appBarTheme.backgroundColor,
+        ),
+        body: Column(
+          children: <Widget>[
+            Expanded(
+                flex: 5,
+                child: ChangeNotifierProvider(
+                    create: (_) => ScannerProvider(),
+                    builder: (context, snapshot) {
+                      return Consumer<ScannerProvider>(
+                          builder: (context, provider, snapshot) {
+                        return QRView(
+                            key: qrKey,
+                            overlay: QrScannerOverlayShape(
+                                borderColor: Theme.of(context).accentColor,
+                                borderRadius: 10,
+                                borderLength: 30,
+                                borderWidth: 10,
+                                cutOutSize: 250),
+                            onQRViewCreated: (controller) {
+                              controller.scannedDataStream
+                                  .listen((scanData) async {
+                                controller.pauseCamera();
+                                await provider.resolveDataScan(scanData);
+                                provider.submitQR();
+                                passToMainScreen = provider.result;
+                                ScaffoldMessenger.of(context).showSnackBar(
+                                  SnackBar(
+                                    backgroundColor:
+                                        Theme.of(context).primaryColor,
+                                    content: Row(
+                                      children: [
+                                        Spacer(),
+                                        Icon(
+                                          Icons.qr_code,
+                                          color: Colors.white,
+                                        ),
+                                        SizedBox(width: 10),
+                                        Text(
+                                          //Displays the format of the QR Code and the resuling text.
+                                          '${provider.format!.toUpperCase()}: ${provider.result!.toUpperCase()}',
+                                          style: TextStyle(
+                                              color: Colors.white,
+                                              fontSize: 16),
+                                        ),
+                                        Spacer()
+                                      ],
+                                    ),
+                                  ),
```

Comment:
bug: The text overflows when the message fetched from the QR code is large. This is a staged example for ease of reproducing but also happens with some of the API responses.
![Screenshot_1630454207](https://user-images.githubusercontent.com/43831668/131590697-142ac720-1400-4669-a576-4c6d8b073fe9.png)

Reply:
Aha. I believe this can be solved by flexing the text widget. I do love to fix the yellow-black lines lol